src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java

```diff
@@ -42,24 +42,34 @@ public class HotRodService implements CacheService {
 
    @Inject DBSyncerMetrics metrics;
 
-   private final Map<URI, HotRodCache> managers = new ConcurrentHashMap<>();
+   private final ConcurrentHashMap<URI, HotRodCache> managers = new ConcurrentHashMap<>();
 
    @Override
    public CompletionStage<Boolean> reconnect(CacheIdentifier identifier, Rule rule) {
       var cache = getCacheOrThrow(identifier);
+      if (!cache.backends.containsKey(identifier.rule())) throw new IllegalStateException(String.format("Backend not found for '%s'", identifier));
+
       return cache.rcm.startAsync().thenApply(ignore -> {
-         var backend = cache.backends.get(identifier.rule());
-         if (backend == null) throw new IllegalStateException(String.format("Backend not found for '%s'", identifier));
-         return backend.reconnect();
+         var offset = new HotRodOffsetBackend(cache.rcm.getCache(OFFSET_CACHE_NAME));
+         var schema = createSchemaBackend(cache.rcm);
+         var newCache = new HotRodCache(cache.rcm, offset, schema);
+
+         var oldBackend = cache.backends.get(identifier.rule());
+         var backend = oldBackend.copy(newCache.rcm.getCache(identifier.rule()));
+         newCache.backends.putAll(cache.backends);
+         newCache.backends.put(identifier.rule(), backend);
+
+         return managers.replace(identifier.uri(), cache, newCache) && backend.reconnect();
```

Comment:
Without creating the backends, I kept getting an exception throwing that the pool was terminated (https://github.com/infinispan/infinispan/blob/main/client/hotrod/src/main/java/org/infinispan/hotrod/impl/transport/netty/ChannelPool.java#L73), even after the `startAsync()` completing successfully.